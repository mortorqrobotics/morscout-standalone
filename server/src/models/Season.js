import { Schema } from "mongoose";

export default mongoose => {
  const SeasonSchema = new Schema({
    year: {
      type: Number,
      validate: year => year >= 1989 && year <= Date.now().getUTCFullYear() + 1,
      default: Date.now
    },
    name: String,
    buildStart: {
      type: Date,
      validate: date =>
        date.getUTCFullYear() === this.year ||
        date.getUTCFullYear() === this.year - 1
    },
    buildEnd: {
      type: Date,
      validate: date =>
        (date.getUTCFullYear() === this.year ||
          date.getUTCFullYear() === this.year - 1) &&
        date.time() > this.buildStart.time()
    },
    seasonStart: {
      type: Date,
      validate: date =>
        date.getUTCFullYear() === this.year &&
        date.time() > this.buildEnd.time()
    },
    seasonEnd: {
      type: Date,
      validate: date => date.time() > this.seasonStart.time()
    }
  });

  return mongoose.model("Season", SeasonSchema);
};
