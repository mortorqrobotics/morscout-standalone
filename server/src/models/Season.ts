import * as mongoose from "mongoose";

const Season: mongoose.Schema = new mongoose.Schema({
  year: {
    type: Number,
    validate: year => year >= 1989 && year <= (new Date()).getUTCFullYear() + 1,
    default: () => (new Date()).getUTCFullYear()
  },
  name: String,
  buildStart: {
    type: Date,
    validate: date =>
      date.getUTCFullYear() === this.year ||
      date.getUTCFullYear() === this.year - 1
  },
  buildEnd: {
    type: Date,
    validate: date =>
      (date.getUTCFullYear() === this.year ||
        date.getUTCFullYear() === this.year - 1) &&
      date.time() > this.buildStart.time()
  },
  seasonStart: {
    type: Date,
    validate: date =>
      date.getUTCFullYear() === this.year && date.time() > this.buildEnd.time()
  },
  seasonEnd: {
    type: Date,
    validate: date => date.time() > this.seasonStart.time()
  }
});

export default mongoose.model("Season", Season);
