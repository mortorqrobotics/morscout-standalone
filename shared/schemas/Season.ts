import * as mongoose from "mongoose";

const Season: mongoose.Schema = new mongoose.Schema({
  year: {
    type: Number,
    validate: year => year >= 1989 && year <= new Date().getUTCFullYear() + 1,
    default: () => new Date().getUTCFullYear(),
    required: true
  },
  name: {
    type: String,
    required: true
  },

  buildStart: {
    type: Date,
    required: true
  },
  buildEnd: {
    type: Date,
    required: true
  },
  seasonStart: {
    type: Date,
    required: true
  },
  seasonEnd: {
    type: Date,
    required: true
  }
});

export default Season;
