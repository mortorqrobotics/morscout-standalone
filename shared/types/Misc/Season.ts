export const loadSeason = "GET_SEASON";
export const updatedSeason = "UPDATED_SEASON";
