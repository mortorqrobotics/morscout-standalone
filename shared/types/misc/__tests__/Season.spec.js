import { loadSeason, updatedSeason } from "../Season";

test("load Team stays the same", () => {
  expect(loadSeason).toMatchSnapshot();
});

test("updated Team stays the same", () => {
  expect(updatedSeason).toMatchSnapshot();
});
