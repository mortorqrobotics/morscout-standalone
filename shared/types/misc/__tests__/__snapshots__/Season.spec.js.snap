// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`load Team stays the same 1`] = `"GET_SEASON"`;

exports[`updated Team stays the same 1`] = `"UPDATED_SEASON"`;
